Automattic/wp-calypso
# Stats: Update component FollowersCount and its link to subscribers

Related to #

## Proposed Changes

* Fix the page link for viewing subscribers on the `StatsNavigation`.
* Rename the component `FollowersCount` to `SubscribersCount`.

## Testing Instructions



* Spin up this change with the Calypso Live link.
* Navigate to the Stats `Traffic` page.
* Ensure the navigation link for directing to the subscribers' page is correct.

<img width="1274" alt="截圖 2023-03-10 下午3 09 20" src="https://user-images.githubusercontent.com/6869813/224247694-393013b8-e187-425b-bbcd-92df90fa583a.png">

## Pre-merge Checklist
